lib/testing.dart

```diff
@@ -10,14 +11,25 @@ import 'package:bazel_worker/bazel_worker.dart';
 export 'src/utils.dart' show protoToDelimitedBuffer;
 
 /// A [Stdin] mock object.
+///
+/// Note: When using this with an [AsyncWorkerLoop] you must call [close] in
+/// order for the loop to exit properly.
 class TestStdinStream implements Stdin {
+  /// Pending bytes to be delivered synchronously.
   final Queue<int> pendingBytes = new Queue<int>();
 
-  // Adds all the input bytes to this stream.
+  /// Controls the stream for async delivery of bytes.
+  final StreamController _controller = new StreamController();
+
+  /// Adds all the [bytes] to this stream.
   void addInputBytes(List<int> bytes) {
     pendingBytes.addAll(bytes);
+    _controller.add(bytes);
```

Comment:
Ya, I was playing a bit fast and loose with this. The code basically assumes you only use it in either an async or sync context. Realistically I think its actually a safe assumption, but if you would like I can separate it out into two classes.